src/components/Balance.tsx

```diff
@@ -44,35 +52,99 @@ export const Balance = ({
     return result;
   };
 
-  const handleDeposit = (asset: Types.AssetBalance) => {
-    // TODO: handle global errors on UI
+  const handleSep24Deposit = (asset: Asset) => {
     dispatch(
       depositAssetAction({
-        assetCode: asset.token.code,
-        assetIssuer: asset.token.issuer.key,
+        assetCode: asset.assetCode,
+        assetIssuer: asset.assetIssuer,
       }),
     );
   };
 
-  const handleWithdraw = (asset: Types.AssetBalance) => {
-    // TODO: handle global errors on UI
+  const handleSep24Withdraw = (asset: Asset) => {
     dispatch(
       withdrawAssetAction({
-        assetCode: asset.token.code,
-        assetIssuer: asset.token.issuer.key,
+        assetCode: asset.assetCode,
+        assetIssuer: asset.assetIssuer,
       }),
     );
   };
 
-  const handleSep31Send = (asset: Types.AssetBalance) => {
+  const handleSep31Send = (asset: Asset) => {
     dispatch(
       fetchSendFieldsAction({
-        assetCode: asset.token.code,
-        assetIssuer: asset.token.issuer.key,
+        assetCode: asset.assetCode,
+        assetIssuer: asset.assetIssuer,
       }),
     );
   };
 
+  const handleActionChange = ({
+    actionId,
+    balance,
+  }: {
+    actionId: string;
+    balance: Asset;
+  }) => {
+    if (!actionId) {
+      return;
+    }
+
+    let props: AssetActionItem | undefined;
+
+    switch (actionId) {
+      case AssetActionId.SEND_PAYMENT:
+        // TODO: title + description
+        props = {
+          id: balance.assetString,
+          balance,
```

Comment:
very small - but would it make sense to declared a `defaultProps` object that = `{ id: balance.Assetstring, balance }` since every case where you're reassigning props, those are included?

For ex: 
```
let props: AssetActionItem | undefined;
const defaultProps = {
  id: balance.assetString,
  balance,
};

    switch (actionId) {
      case AssetActionId.SEND_PAYMENT:
        // TODO: title + description
        props = {
          ...defaultProps,
          title: `Send payment ${balance.assetCode}`,
          description: "Send payment description",
          callback: onSend,
        }

```

Would be helpful if in the future every case needs a new prop or one of these default props needs to be changed in some way

Reply:
I like that, good point. 👍